src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -199,9 +200,11 @@ private[snowflake] class JDBCWrapper {
            |""".stripMargin.filter(_ >= ' '))
     }
 
-    // Important: Set "snowflake.client.info" is very important!
+    // Important: Set client_info is very important!
     // For more details, refer to PROPERTY_NAME_OF_CONNECTOR_VERSION
-    System.setProperty("snowflake.client.info", Utils.getClientInfoString())
+    // NOTE: From JDBC 3.13.3, the client info can be set with JDBC properties
+    // instead of system property: "snowflake.client.info".
+    jdbcProperties.put(SFSessionProperty.CLIENT_INFO.getPropertyKey, Utils.getClientInfoString())
```

Comment:
To reviewers:
`SFSessionProperty.CLIENT_INFO` is a new property from JDBC 3.13.3, which is used to set CLIENT INFO with JDBC properties.

Reply:
So, that is a JDBC change, we just apply this change?

Reply:
Now, JDBC will check this CLIENT_INFO first. If it is not, it checks System property so that some old spark connector can use new JDBC.

The story is:
1. We found this problem and discuss with JDBC team. https://snowflake.slack.com/archives/GCA15K0QH/p1618940626292300
2. They change JDBC API to support set CLIENT INFO with JDBC properties.
3. Now, we change SC to use the new API.